### PR TITLE
[Supplier] refactor: supplier type to shared protobuf package

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

image

## Issue

Related to #54 

This resolves the dependency cycle between the `pocket.supplier` & `pocket.session` protobuf packages by introducing a new `pocket.shared` package. `pocket.session.Session` depends  on `Supplier`, and `pocket.supplier.MsgCreateClaim` (& `MsgSubmitProof`) depend on `pocket.session.SessionHeader`.

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [x] Other (specify): Refactor

## Testing

- [x] **Run all unit tests**: `make go_test`
- [x] **Verify Localnet manually**: See the instructions [here](TODO: add link to instructions)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

**Review by Olshansk — DISMISSED** [review]
Let's give it a shot!

File: proto/pocket/session/session.proto [hunk 1]
```diff
@@ -7,7 +7,7 @@ import "cosmos_proto/cosmos.proto";
 // TODO(@Olshansk): Uncomment the line below once the service.proto file is added
 // import "pocket/service/service.proto";
 import "pocket/application/application.proto";
-import "pocket/shared/supplier.proto";
+import "pocket/supplier/supplier.proto";
```
Comment (bryanchriswhite) [inline proto/pocket/session/session.proto:10]:
```suggestion
import "pocket/shared/supplier.proto
```

File: proto/pocket/session/session.proto [hunk 2]
```diff
@@ -17,7 +17,7 @@ message SessionHeader {
     string application_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // The address of the application this session is for
     // TODO(@Olshansk): Uncomment the line below once the `ServiceId` proto is defined
     // service.ServiceId service_id = 2; // The ID of the service this session is servicing
-    int64 session_block_start_height = 3; // The height at which this session started
+    int64 session_start_block_height = 3; // The height at which this session started
```
Comment (bryanchriswhite) [inline proto/pocket/session/session.proto:20]:
```suggestion
    int64 session_block_start_height = 3; // The height at which this session started
```

File: proto/pocket/session/session.proto [hunk 3]
```diff
@@ -32,5 +32,5 @@ message Session {
     // TODO(@Olshansk): Uncomment the line below once the `Service` proto is defined
     // service.Service service = 5; // A fully hydrated service object this session is for
     application.Application application = 6; // A fully hydrated application object this session is for
-    repeated pocket.shared.Supplier suppliers = 7; // A fully hydrated set of servicers that are serving the application
+    repeated supplier.Supplier suppliers = 7; // A fully hydrated set of servicers that are serving the application
```
Comment (bryanchriswhite) [inline proto/pocket/session/session.proto:35]:
```suggestion
    repeated pocket.shared.Supplier suppliers = 7; // A fully hydrated set of servicers that are serving the application
```

**Review by Olshansk — APPROVED** [review]

**Pull request merged**